### PR TITLE
Update .NET Tool package type

### PR DESCRIPTION
**Pull request by loic-sharma**

### CODE REVIEW

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[4a0e15e](https://github.com/NuGet/docs.microsoft.com-nuget/commits/4a0e15ebb254942f4b78d756dd760f614807237b)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-89d0d76b)

<a id="89d0d76b"></a>
###
 - **Line 0**: **[Error]** `````Failure to clone the repository `https://github.com/Microsoft/templates.docs.msft#master`.This could be caused by an incorrect repository URL, please verify the URL on the Docs Portal (https://ops.microsoft.com).This could also be caused by not having the proper permission the repository, please confirm that the GitHub group/team that triggered the build has access to the repository.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C1%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202001282148522365-1860%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=1G%2FEifI3xRHTEi6aHCJPoceZBC1vCbeq453X2xziZ1g%3D&st=2020-01-28T21%3A44%3A22Z&se=2020-02-28T21%3A49%3A22Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/create-packages/set-package-type.md [hunk 1]
```diff
@@ -13,9 +13,9 @@ With NuGet 3.5+, packages can be marked with a specific *package type* to indica
 
 - `Dependency` type packages add build- or run-time assets to libraries and applications, and can be installed in any project type (assuming they are compatible).
 
-- `DotnetCliTool` type packages are extensions to the [dotnet CLI](/dotnet/articles/core/tools/index) and are invoked from the command line. Such packages can be installed only in .NET Core projects and have no effect on restore operations. More details about these per-project extensions are available in the  [.NET Core extensibility](/dotnet/articles/core/tools/extensibility#per-project-based-extensibility) documentation.
+- `DotnetTool` type packages are extensions to the [dotnet CLI](/dotnet/articles/core/tools/index) and are invoked from the command line. Such packages can be installed only in .NET Core projects and have no effect on restore operations. More details about these per-project extensions are available in the  [.NET Core extensibility](/dotnet/articles/core/tools/extensibility#per-project-based-extensibility) documentation.
```
Comment (joelverhagen) [inline docs/create-packages/set-package-type.md]: Seems fine, but it's a little strange to name package types that we don't own the scenario around. What about "Template" and the SDK package type (forget the name)?

Comment (loic-sharma) [reply docs/create-packages/set-package-type.md]: The most popular package types appear to be:

Package type | Occurrence
-- | --
DotnetTool | 6590
Template | 4046
DotnetCliTool | 1209
MSBuildSdk | 758
Dependency | 526
AzureSiteExtension | 463
DotnetPlatform | 213

Given this data, it seems that MSBuildSDk, AzureSiteExtensions, and DotnetPlatform are rather esoteric types. I'll add a quick blurb for templates though

Comment (joelverhagen) [reply docs/create-packages/set-package-type.md]: Is it sustainable for us to keep maintaining this list?

Like what does this list even mean? Is it "package types that Microsoft ships"? Or is it ones that nuget.org recognizes and changes the command palette for?

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[c047ab6](https://github.com/NuGet/docs.microsoft.com-nuget/commits/c047ab6027a9e9797d4c169d482aa11b71a24b54)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/create-packages/set-package-type.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/loic-sharma-patch-2/docs/create-packages/set-package-type.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/create-packages/set-package-type?branch=pr-en-us-1860) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C1%5C31%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202001310030302897-1860%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=K1MHQScDlW9pZOJ7i8kNm2LW0XeADmNn3%2FCqI0FyIlk%3D&st=2020-01-31T00%3A26%3A06Z&se=2020-03-02T00%3A31%3A06Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/create-packages/set-package-type.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ With NuGet 3.5+, packages can be marked with a specific *package type* to indica
 
 - `DotnetTool` type packages are extensions to the [dotnet CLI](/dotnet/articles/core/tools/index) and are invoked from the command line. Such packages can be installed only in .NET Core projects and have no effect on restore operations. More details about these per-project extensions are available in the  [.NET Core extensibility](/dotnet/articles/core/tools/extensibility#per-project-based-extensibility) documentation.
 
+- `Template` type packages provide [custom templates](/dotnet/core/tools/custom-templates) that can be used to create files or projects like an app, service, tool, or class library.
```
Comment (joelverhagen) [inline docs/create-packages/set-package-type.md]: Make sure this link works by clicking it on the test deployment -- I've had these relative links act weird before.

**Pull request merged**